controllers/ovndbcluster_controller.go

```diff
@@ -760,23 +755,28 @@ func (r *OVNDBClusterReconciler) reconcileServices(
 			}
 
 			dnsIP, err := getPodIPInNetwork(ovnPod, instance.Namespace, instance.Spec.NetworkAttachment)
+			dnsIPsList = append(dnsIPsList, dnsIP)
 			if err != nil {
 				return ctrl.Result{}, err
 			}
 
-			// Create DNSData CR
-			err = ovndbcluster.DNSData(
-				ctx,
-				helper,
-				serviceName,
-				dnsIP,
-				instance,
-				ovnPod,
-				serviceLabels,
-			)
-			if err != nil {
-				return ctrl.Result{}, err
-			}
+		}
+		dnsDataName := ovnv1.SBDBDNSName
+		if instance.Spec.DBType == ovnv1.NBDBType {
+			dnsDataName = ovnv1.NBDBDNSName
+		}
+		// Create DNSData CR
+		err = ovndbcluster.DNSData(
+			ctx,
+			helper,
+			serviceName,
+			dnsIPsList,
+			instance,
+			dnsDataName,
```

Comment:
Instead of dnsDataName i think we can just use serviceName and avoid all the logic above.